### PR TITLE
Updated DeviceSpec docstring and fixed typos

### PR DESCRIPTION
**Pull request by nikochiko**
Updated docstring (as per #34124), adding instructions for using `device_spec.to_string()` method with eager execution enabled. Fixed missing parentheses in example.

### CODE REVIEW

**Comment by nikochiko** [comment]
@mihaimaruseac Please review

File: tensorflow/python/framework/device_spec.py [hunk 1]
```diff
@@ -47,38 +47,55 @@ def _as_device_str_or_none(device_type):
 @tf_export("DeviceSpec", v1=[])
 class DeviceSpecV2(object):
   """Represents a (possibly partial) specification for a TensorFlow device.
-
+
```
Comment (mihaimaruseac) [inline tensorflow/python/framework/device_spec.py]: Please remove trailing whitespace

File: tensorflow/python/framework/device_spec.py [hunk 2]
```diff
@@ -47,38 +47,55 @@ def _as_device_str_or_none(device_type):
...
   ```
-
+  
+  With eager execution disabled (by default in TensorFlow 1.x and by calling
+  disable_eager_execution() in TensorFlow 2.x), the following syntax
+  can be used:
+  
+  ```python
+  # Location in TensorFlow 2.x
+  from tensorflow.python.framework.ops import disable_eager_execution
```
Comment (mihaimaruseac) [inline tensorflow/python/framework/device_spec.py]: `tf.compat.v1.disable_eager_execution` is the recommended API.

**Review by mihaimaruseac — COMMENTED** [review]
Thank you. Some nits, but it's ok

File: tensorflow/python/framework/device_spec.py (hunk 2 above)
Comment (nikochiko) [reply tensorflow/python/framework/device_spec.py]: Yeah I saw that. But I thought it would be better to avoid the v1 API. Will update it right away.

**Comment by nikochiko** [comment]
Done.

File: tensorflow/python/framework/device_spec.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ def _as_device_str_or_none(device_type):
 @tf_export("DeviceSpec", v1=[])
 class DeviceSpecV2(object):
   """Represents a (possibly partial) specification for a TensorFlow device.
-
+
```
Comment (mihaimaruseac) [inline tensorflow/python/framework/device_spec.py]: This still shows up as a trailing whitespace

```
   """Represents a (possibly partial) specification for a TensorFlow device.
-
+  
   `DeviceSpec`s are used throughout TensorFlow to describe where state is stored
```

(from https://patch-diff.githubusercontent.com/raw/tensorflow/tensorflow/pull/34162.diff)

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by rthadur** [comment]
@nikochiko can you please fix the build failures ?

**Comment by mihaimaruseac** [comment]
```
FAIL: Found 1 non-whitelisted pylint errors:
tensorflow/python/framework/device_spec.py:88: [C0301(line-too-long), ] Line too long (88/80)
```

**Comment by nikochiko** [comment]
Done. Looks good?

**Review by mihaimaruseac — APPROVED** [review]

**Comment by nikochiko** [comment]
Any idea why the Android build is failing?

**Pull request merged**